### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by pataquets**
**What issue type does this pull request address?** (keep at least one, remove the others)  
/kind enhancement
/kind feature

**What is this pull request for? Which issues does it resolve?** (use `resolves #<issue_number>` if possible)  
Enables Docker workflow to build and run the application. Follows build instructions on docs (even runs tests).
* Allows building and running (even hacking) right away without need to install Golang.
* Optimized for repeated builds when developing by separating build steps (ie. deps downloading).
* Not optimized for size but speed, although it can be done, ideally on a child image.
* Avoids resorting to untrusted and/or not up-to-date, third party Docker images by allowing to build it from source yourself.
* Enabler for later adding public Docker image as new distribution/release method (optional).
* Helps document build process.

**Does this pull request has user-facing changes?** (e.g. config changes, new/modified commands, new/modified flags)  
No.

**Does this pull request add new dependencies?**  
Docker, optional.

**What else do we need to know?**  
Uses official `golang` public Docker image as base. More info at https://hub.docker.com/_/golang .
As any Docker image needing to interact with local files, it needs to mount credentials and image dirs using appropriate `-v` switches.
You can test the build result using my Docker Hub public image by doing:
```
$ docker run --rm -it [-v /local/dir/mounts/:/container/dir/mounts/ ...] pataquets/gphotos-uploader-cli-src [command options]
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /go/src/gphotos-uploader-cli/
+COPY go.mod .
+COPY go.sum .
+RUN go mod download -x
+
+COPY . .
+RUN \
+  make test && \
+  make build && \
+  make clean
```
Comment (pacoorozco) [inline Dockerfile:12]: Doesn't `make clean` remove the `gphotos-uploader-cli`?

File: README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ Command line tool to mass upload media folders to your Google Photos account.
 
 # Google Photos uploader CLI   
 
-While the official tool only supports Mac OS and Windows, this brings an uploader to Linux too. Lets you upload photos from, in theory, any OS for which you can compile a Go program.     
+While the official tool only supports Mac OS and Windows, this brings an uploader to Linux too. Lets you upload photos from, in theory, any OS for which you can compile a Go program or build a Docker image using the provided `Dockerfile`.
```
Comment (pacoorozco) [inline README.md:14]: Should we add some documentation about the dockered version? 

You have include it on the PR, should we include it on the documentation too?

File: Dockerfile (hunk 1 above)
Comment (pacoorozco) [reply Dockerfile:12]: Forget it! I was testing it in a different version 😓

**Comment by pacoorozco** [comment]
btw, have you seen [this project](https://github.com/rfgamaral/docker-gphotos-uploader)?

File: Dockerfile (hunk 1 above)
Comment (pataquets) [reply Dockerfile:12]: For your own peace of mind, I already tested it specifically before submitting. 😄

File: README.md (hunk 2 above)
Comment (pataquets) [reply README.md:14]: I wasn't sure on how much to document and I settled on minimal.
It depends on if you want to go further with Docker workflow, ie creating an Automated Build, etc. For people used to work with Docker, just having the Dockerfile handy will be all what they'll need, thou.
Anyway, just let me know how much you think is needed. Or let's postpone it after creating an AB. Your choice.

**Comment by pataquets** [comment]
Yes, thanks for pointing out. That's why I've submitted a Dockerfile. To allow building from source yourself.
The linked image is not up to date, and it's quite opinionated (eg, use S6).
Also, you can easily set an Automated Build on Docker Hub and have a daily build from source on each commit automatically.
See https://docs.docker.com/docker-hub/builds/

File: README.md (hunk 2 above)
Comment (pataquets) [reply README.md:14]: @pacoorozco Any thoughts on how should I continue on this?

**Comment by pacoorozco** [comment]
Sorry @pataquets I want to be sure I understand the benefits of providing a docker image to run the `gphotos-uploader-cli`. 

> Allows building and running (even hacking) right away without need to install Golang.
It seems that you point the docker as a way to develop the CLI wo/ installing golang in your system. So... *is the benefit of the docker to help developers?* What would it happen w/ IDE? 

> Optimized for repeated builds when developing by separating build steps (ie. deps downloading).
This is the same than installing go in your system. Or even more, using the `golang` docker image to build. It seems, as previously, that you are trying to help developer of the CLI. Am I right?

> Avoids resorting to untrusted and/or not up-to-date, third party Docker images by allowing to build it from source yourself.
I'm not sure about what you are referring to. We provide a binary CLI. Are you trying to facilitate users to build their own binary to avoid trusting in the provided one?

File: README.md (hunk 2 above)
Comment (pacoorozco) [reply README.md:14]: It maybe use [multi-stage dockers](https://tutorialedge.net/golang/go-multi-stage-docker-tutorial/) to reduce the size of the final image.

**Pull request closed without merging**